From f9e0053eab8ed2d270c080f2d346fe5c84748f23 Mon Sep 17 00:00:00 2001
From: Hedayat Vatankhah <hedayatv@gmail.com>
Date: Sat, 2 Jun 2012 20:24:31 +0430
Subject: [PATCH 05/18] Detect ld.so after usr move (#826754)

Handle ld.so path changing from /lib to /usr/lib.

REBASE CONFLICT

Resolves: rhbz#826754
---
 os-probes/mounted/common/90linux-distro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os-probes/mounted/common/90linux-distro b/os-probes/mounted/common/90linux-distro
index 6678f22..583a9ec 100755
--- a/os-probes/mounted/common/90linux-distro
+++ b/os-probes/mounted/common/90linux-distro
@@ -17,7 +17,7 @@ type="$3"
 # symlinks we need to also check in $dir/usr/lib* for distributions that
 # moved /lib* to /usr and only left symlinks behind.
 # TODO: look for ld-linux.so on arches that have it
-if (ls "$dir"/lib*/ld*.so* || ls "$dir"/usr/lib*/ld*.so*) >/dev/null 2>/dev/null; then
+if (ls "$dir"/lib*/ld*.so*  && [ -d "$dir/boot" ] || ls "$dir"/usr/lib*/ld*.so*) >/dev/null 2>/dev/null; then
 	if [ -e "$dir/etc/debian_version" ]; then
 		short="Debian"
 		long="$(printf "Debian GNU/Linux (%s)\n" "$(cat "$dir/etc/debian_version")")"
-- 
2.5.5

